### PR TITLE
Resource item

### LINKED ISSUE #1768: [List filters] Update ResourceList examples to use Filters component

### LINKED ISSUE #1769: [List filters] Remove FilterControl docs

### LINKED ISSUE #1770: [List filters] Move ResourceItem to its own component

### LINKED ISSUE #1771: [List filters] Deprecate FilterControl

### LINKED ISSUE #1772: [ResourceItem] Update documentation
Depends on #1770

### LINKED ISSUE #1951: [Filters] additional actions for filters
## Feature request summary

currently it's not possible to add additional actions to `Filters` component like what [FilterControl](https://github.com/Shopify/polaris-react/blob/7e4e9e88007303782a8022b60224e4f9dd8d5cd0/src/components/ResourceList/components/FilterControl/FilterControl.tsx#L19) has.
I basically need it for letting user save their filters but if it's not what `Filters` is intended for, docs should clarify it when `FilterControl` should be used instead of `Filters`?

---

🌟 Feature requests that are not yet planned will be closed. We then use the issue’s [:+1: upvotes](https://help.github.com/articles/about-conversations-on-github/) to track and set priorities. See the [contribution guidelines](https://github.com/Shopify/polaris-react/blob/master/.github/CONTRIBUTING.md) for more information.

### PR DESCRIPTION
**Pull request by tmlayton**
### WHY are these changes introduced?

Closes #1768 
Closes #1769
Closes #1770 
Closes #1771
Closes #1772
Closes #1951 

### WHAT is this pull request doing?

- Moves `ResourceItem` to its own component (formerly called `Item` and a subcomponent of `ResourceList`). The component is still available at `ResourceList.Item`
- Updates the `ResourceItem` docs per [the new documentation](https://docs.google.com/document/d/1wuM0bBexvJQvmUsk0EBCbCjyiLxEFz8hdxCYJY3UFNE/edit?ts=5d2c9226)
- Removes the `FilterControl` docs
- Deprecates `FilterControl`
- Updates `ResourceList` examples to use the `Filters` component
- Simplifies `ResourceList` docs and removes case studies for now until they can find a new home in the style guide
- Adds example to `Filters` which uses children content
- Used `inlineLabel` for the sort select in `ResourceList`

### 📸

|Resource item|Resource list|
|---|---|
|screencapture-polaris-myshopify-io-components-lists-and-tables-resource-item-2019-08-16-17_42_46|screencapture-polaris-myshopify-io-components-lists-and-tables-resource-list-2019-08-16-17_44_47|

### 🎩 checklist

* [x] Tested on [mobile](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting.md#cross-browser-testing)
* [x] Tested on [multiple browsers](https://help.shopify.com/en/manual/intro-to-shopify/shopify-admin/supported-browsers)
* [x] Tested for [accessibility](https://github.com/Shopify/polaris-react/blob/master/documentation/Accessibility%20testing.md)
* [x] Updated the component's `README.md` with documentation changes
* [x] [Tophatted documentation](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting%20documentation.md) changes in the style guide

### CODE REVIEW

File: src/components/ResourceItem/README.md [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+        </ResourceItem>
+      );
+    }}
+  />
+</Card>
+```
+
+### Item with media
+
+The media element can hold an [avatar](/components/images-and-icons/avatar), [thumbnail](/components/images-and-icons/thumbnail) or other small-format graphic.
```
Comment (claydelk) [inline src/components/ResourceItem/README.md]:
```suggestion
The media element can hold an [avatar](/components/images-and-icons/avatar), [thumbnail](/components/images-and-icons/thumbnail), or other small-format graphic.
```

File: src/components/ResourceItem/README.md [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+        </ResourceItem>
+      );
+    }}
+  />
+</Card>
+```
+
+### Item with shortcut actions
+
+Shortcut actions present popular actions from the resource’s details page for easy access. A shortcut action should available on every item in the list.
```
Comment (claydelk) [inline src/components/ResourceItem/README.md]:
```suggestion
Shortcut actions present popular actions from the resource’s details page for easy access. A shortcut action should be available on every item in the list.
```

File: src/components/ResourceItem/README.md [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+  - list attributes
+  - exceptions list
+  - list secondary actions
+  - secondary actions in a list
+  - list of resources
+---
+
+# Resource item
+
+Resource items represent specific objects within a collection, like products or orders. They provide contextual information on the resource type and link to the object’s detail page.
```
Comment (andyhealey26) [inline src/components/ResourceItem/README.md]: "such as" or "for example" rather than "like"

File: src/components/ResourceItem/README.md [hunk 4]
```diff
@@ -0,0 +1,212 @@
...
+---
+## Required components
+
+The resource item component must be wrapped in the [resource list](/components/lists-and-tables/resource-list) component.
+
+---
+
+## Accessibility
+
+Resource items function as links to the full-page representations of the items. Each item should have a unique `name` prop. For each `ResourceItem`, the `accessibilityLabel` prop should be used to give the link a unique `aria-label` value. The `aria-label` value should convey the link’s purpose, using the `name` value. Merchants who use screen readers or other text to speech tools should be able to easily distinguish each link from the others.
```
Comment (andyhealey26) [inline src/components/ResourceItem/README.md]: Not super clear what this means. "Function as" or "are"?

Comment (andyhealey26) [inline src/components/ResourceItem/README.md]: "prop"?

Comment (andyhealey26) [reply src/components/ResourceItem/README.md]: Maybe swop out "should" to make it more active, eg "Use the aria-label to convey the link's purpose"
(optional)

File: src/components/ResourceItem/README.md [hunk 5]
```diff
@@ -0,0 +1,212 @@
...
+
+When adding custom content to resource items, ensure that all text is available to all users and that all custom controls have a unique accessible name to help users understand what will happen when the control is activated.
+
+### Keyboard
+
+Links can be activated with the <kbd>enter</kbd>/<kbd>return</kbd> key by default.
+
+If you add custom controls to resource items, then ensure that the controls:
```
Comment (andyhealey26) [inline src/components/ResourceItem/README.md]: "make sure" is easier than "ensure"

Comment (andyhealey26) [reply src/components/ResourceItem/README.md]: Period after "accessible name" then new sentence "This helps user to understand..."

File: src/components/ResourceItem/README.md [hunk 6]
```diff
@@ -0,0 +1,212 @@
...
+- Can be used with the keyboard
+- Receive keyboard focus in a logical order
+- Display a visible focus indicator
+
+---
+## Best practices
+
+Resource items should:
+
+- Be tailored to the specific type of context being displayed.
```
Comment (andyhealey26) [inline src/components/ResourceItem/README.md]: Can this be clearer? like maybe with an example?

File: src/components/ResourceItem/README.md [hunk 7]
```diff
@@ -0,0 +1,212 @@
...
+- Perform an action when clicked. The action should navigate to the resource’s details page or otherwise provide more detail about the item.
```
Comment (andyhealey26) [inline src/components/ResourceItem/README.md]: Delete "otherwise"

File: src/components/ResourceItem/README.md [hunk 8]
```diff
@@ -0,0 +1,212 @@
...
+
+- Provide [shortcut actions](/components/lists-and-tables/resource-list#study-custom-item-shortcut-actions) for quick access to frequent actions from the resource’s details page.
+
+---
+
+## Content guidelines
+
+Resource items should:
+
+- Present the content merchants need to find the items that they’re looking for.
```
Comment (andyhealey26) [inline src/components/ResourceItem/README.md]: Should this say "context"? If not then the sentence isn't clear

File: src/components/ResourceItem/README.md [hunk 9]
```diff
@@ -0,0 +1,212 @@
...
+- Support merchant tasks for the particular type of resource.
+- Avoid colons.
```
Comment (andyhealey26) [inline src/components/ResourceItem/README.md]: Just colons or anything else?

File: src/components/ResourceItem/README.md (hunk 3 above)
Comment (VanjaMitrovic) [inline src/components/ResourceItem/README.md]:
```suggestion
Resource items represent specific objects within a collection, such as products or orders. They provide contextual information on the resource type and link to the object’s detail page.
```

File: src/components/ResourceItem/README.md [hunk 10]
```diff
@@ -0,0 +1,212 @@
...
+
+Resource items function as links to the full-page representations of the items. Each item should have a unique `name` prop. For each `ResourceItem`, the `accessibilityLabel` prop should be used to give the link a unique `aria-label` value. The `aria-label` value should convey the link’s purpose, using the `name` value. Merchants who use screen readers or other text to speech tools should be able to easily distinguish each link from the others.
+
+When adding custom content to resource items, ensure that all text is available to all users and that all custom controls have a unique accessible name to help users understand what will happen when the control is activated.
+
+### Keyboard
+
+Links can be activated with the <kbd>enter</kbd>/<kbd>return</kbd> key by default.
+
+If you add custom controls to resource items, then ensure that the controls:
```
Comment (VanjaMitrovic) [inline src/components/ResourceItem/README.md]:
```suggestion
If you add custom controls to resource items, then make sure that the controls:
```

File: src/components/ResourceItem/README.md [hunk 11]
```diff
@@ -0,0 +1,212 @@
...
+- Receive keyboard focus in a logical order
+- Display a visible focus indicator
+
+---
+## Best practices
+
+Resource items should:
+
+- Be tailored to the specific type of context being displayed.
+- Perform an action when clicked. The action should navigate to the resource’s details page or otherwise provide more detail about the item.
```
Comment (VanjaMitrovic) [inline src/components/ResourceItem/README.md]:
```suggestion
- Perform an action when clicked. The action should navigate to the resource’s details page or provide more detail about the item.
```

File: src/components/ResourceItem/README.md [hunk 12]
```diff
@@ -0,0 +1,212 @@
...
+
+- Provide [shortcut actions](/components/lists-and-tables/resource-list#study-custom-item-shortcut-actions) for quick access to frequent actions from the resource’s details page.
+
+---
+
+## Content guidelines
+
+Resource items should:
+
+- Present the content merchants need to find the items that they’re looking for.
```
Comment (VanjaMitrovic) [inline src/components/ResourceItem/README.md]:
```suggestion
- Present the information that merchants need to find the items that they’re looking for.
```

**Comment by jamieroyce** [comment]
Is there any way to include more documentation on HOW to use ResourceList filters with GraphQL. This is not easy for a newcomer to learn.  See my post to the community [here.](https://community.shopify.com/c/Shopify-APIs-SDKs/ResourceList-Filter-Example-needed-with-actual-data-that-works/m-p/843563)

I have a GraphQL query ready with a $filter input variable. Just need help on how to connect the ResourceList to the query as users type in the filter field.

**Comment by tmlayton** [comment]
@jamieroyce I’d recommend opening an issue for this.

**Comment by jamieroyce** [comment]
Thanks and will do :)

**Comment by jamieroyce** [comment]
https://github.com/Shopify/polaris-react/issues/3186

Thank you for the advice.

**Pull request merged**